src/Sinch/Fax/Services/ServicesClient.cs

```diff
@@ -0,0 +1,301 @@
+            uriBuilder.Path += $"services/{serviceId}/numbers/{phoneNumber}/emails";
+            var queryString = HttpUtility.ParseQueryString(string.Empty);
+            if (page.HasValue)
+            {
+                queryString.Add("page", page.Value.ToString());
+            }
+
+            if (pageSize.HasValue)
+            {
+                queryString.Add("pageSize", pageSize.Value.ToString());
+            }
+
+            uriBuilder.Query = queryString.ToString();
+            return _http.Send<ListEmailsResponse<string>>(uriBuilder.Uri, HttpMethod.Get, cancellationToken);
+        }
+
+        public async IAsyncEnumerable<string> ListEmailsForNumberAuto(string serviceId, string phoneNumber, int? page,
+            int? pageSize,
+            [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            _logger?.LogDebug("Auto Listing emails");
+
+            ListEmailsResponse<string> response;
+            do
+            {
+                response = await ListEmailsForNumber(serviceId, phoneNumber, page, pageSize, cancellationToken);
+                foreach (var contact in response.Emails)
+                    yield return contact;
+                page += 1;
+            } while (!Utils.IsLastPage(response.PageNumber, response.PageSize, response.TotalItems, PageStart.One));
+        }
+
+        public Task<ListNumbersResponse> ListNumbers(string serviceId, int? page, int? pageSize,
+            CancellationToken cancellationToken = default)
+        {
+            _logger?.LogInformation("Listing numbers for {service}...", serviceId);
+
+            ExceptionUtils.CheckEmptyString(nameof(serviceId), serviceId);
+
+            var uriBuilder = new UriBuilder(_apiBasePath);
+
+            uriBuilder.Path += $"/{serviceId}/numbers";
+
+            var queryString = HttpUtility.ParseQueryString(string.Empty);
+            if (page.HasValue)
+            {
+                queryString.Add("page", page.Value.ToString());
+            }
+
+            if (pageSize.HasValue)
+            {
+                queryString.Add("pageSize", pageSize.Value.ToString());
+            }
+
+            uriBuilder.Query = queryString.ToString();
+
+            return _http.Send<ListNumbersResponse>(uriBuilder.Uri, HttpMethod.Get, cancellationToken);
+        }
+
+        public async IAsyncEnumerable<ServicePhoneNumber> ListNumbersAuto(string serviceId, int? page,
+            int? pageSize,
+            [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            _logger?.LogDebug("Auto Listing numbers for {service}", serviceId);
+
+            ExceptionUtils.CheckEmptyString(nameof(serviceId), serviceId);
+
+
+            ListNumbersResponse response;
+            do
+            {
+                response = await ListNumbers(serviceId, page, pageSize, cancellationToken);
+                foreach (var number in response.PhoneNumbers)
```

Comment:
it should be
```suggestion
                foreach (var number in response.Numbers)
```
according to https://developers.sinch.com/docs/fax/api-reference/fax/tag/Services/#tag/Services/operation/listNumbersForService!c=200&path=numbers&t=response

Reply:
For the sake of consistency response class will be the same, but internally it will be handled by `numbers`
I've notified fax team about that inconsistency